### PR TITLE
add Mainnet Guardian Upgrade and SC Changes tasks

### PR DESCRIPTION
**Pull request by mds1**
Changes:
- Bumps base-contracts version to the latest, which incorporates https://github.com/base-org/contracts/pull/84.
- The 010-1 task is complete with CI passing.
- The 010-2 task is complete with CI passing.
- The 009 task still is passing CI, meaning the changes in this PR did not break it

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -244,3 +266,4 @@ workflows:
       - just_simulate_sc_rehearsal_4
       - just_simulate_oeth-002-update-l2pao-to-aliased-l1pao
       - just_simulate_009-fp-upgrade
+      - just_simulate_eth-010-1-guardian-upgrade
```
Comment (maurelian) [inline .circleci/config.yml:269]: Also need 010-2

Comment (mds1) [reply .circleci/config.yml:287]: Done in 0317a5a

File: tasks/eth/010-1-guardian-upgrade/VALIDATION.md [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+### `0x95703e0982140D16f8ebA6d158FccEde42f04a4C` (`SuperchainConfig`)
+
+Links:
+- [Etherscan](https://etherscan.io/address/0x95703e0982140D16f8ebA6d158FccEde42f04a4C)
+
+State Changes:
+
+- **Key:** 0xd30e835d3f35624761057ff5b27d558f97bd5be034621e62240e5c0b784abe68<br/>
+  **Before:** 0x0000000000000000000000009ba6e03d8b90de867373db8cf1a58d2f7f006b3a<br/>
+  **After:** 0x0000000000000000000000007a50f00e8D05b95F98fE38d8BeE366a7324dCf7E<br/>
```
Comment (maurelian) [inline tasks/eth/010-1-guardian-upgrade/VALIDATION.md:132]:
```suggestion
  **After:** 0x00000000000000000000000009f7150D8c019BeF34450d6920f6B3608ceFdAf2<br/>
```

File: tasks/eth/010-1-guardian-upgrade/README.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+## State Validations
+
+Please see the instructions for [validation](./VALIDATION.md).
+
+## Execution
+
+At this point you may resume following the execution instructions in the "Execute the Transaction" section of [NESTED.md](../../../NESTED.md).
+
+When executing, ensure the logs say `Using script Using script /your/path/to/superchain-ops/tasks/eth/010-1-guardian-upgrade/NestedSignFromJson.s.sol`. This ensures all safety checks are run. If the default `NestedSignFromJson.s.sol` script is shown (without the full path), something is wrong and the safety checks will not run.
```
Comment (maurelian) [inline tasks/eth/010-1-guardian-upgrade/README.md:39]:
```suggestion
When executing, ensure the logs say `Using script /your/path/to/superchain-ops/tasks/eth/010-1-guardian-upgrade/NestedSignFromJson.s.sol`. This ensures all safety checks are run. If the default `NestedSignFromJson.s.sol` script is shown (without the full path), something is wrong and the safety checks will not run.
```

File: tasks/eth/010-1-guardian-upgrade/NestedSignFromJson.s.sol [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+        }
+
+        // And finally, we set the Safe signer exceptions.
+        address[] memory shouldHaveCodeExceptions = new address[](numberOfSafeSignersWithNoCode);
+        for (uint256 i = 0; i < safeSignersWithNoCode.length; i++) {
+            shouldHaveCodeExceptions[i] = safeSignersWithNoCode[i];
+        }
+
+        return shouldHaveCodeExceptions;
+    }
```
Comment (maurelian) [inline tasks/eth/010-1-guardian-upgrade/NestedSignFromJson.s.sol:167]: I think we can remove this whole function?

**Review by maurelian — COMMENTED** [review]
Leaving comments on 010-1.
Have not reviewed 010-2 yet.

File: tasks/eth/010-1-guardian-upgrade/VALIDATION.md [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+  **Value:** 0x0000000000000000000000000000000000000000000000000000000000000001 <br/>
+  **Meaning:** The threshold is set to 1. The key can be validated by the location of the `threshold` variable in the [Safe's Storage Layout](https://github.com/safe-global/safe-smart-account/blob/186a21a74b327f17fc41217a927dea7064f74604/contracts/examples/libraries/GnosisSafeStorage.sol#L14).
+
+- **Key:** 0x0000000000000000000000000000000000000000000000000000000000000005 <br/>
+  **Value:** 0x0000000000000000000000000000000000000000000000000000000000000003 <br/>
+  **Meaning:** Sets the nonce to 3 for simulation and signing, since this is the expected nonce of the Safe at the time of execution. The key can be validated by the location of the `nonce` variable in the [Safe's Storage Layout](https://github.com/safe-global/safe-smart-account/blob/186a21a74b327f17fc41217a927dea7064f74604/contracts/examples/libraries/GnosisSafeStorage.sol#L17).
+
+The following two overrides are modifications to the [`owners` mapping](https://github.com/safe-global/safe-contracts/blob/v1.4.0/contracts/libraries/SafeStorage.sol#L15). For the purpose of calculating the storage, note that this mapping is in slot `2`.
```
[the remark this answers is not in the document]
Comment (mds1) [reply tasks/eth/010-1-guardian-upgrade/VALIDATION.md]: Fixed in 59db70b

File: tasks/eth/010-1-guardian-upgrade/VALIDATION.md (hunk 2 above)
Comment (mds1) [reply tasks/eth/010-1-guardian-upgrade/VALIDATION.md]: Fixed in cbecc39

File: tasks/eth/010-1-guardian-upgrade/README.md (hunk 3 above)
Comment (mds1) [reply tasks/eth/010-1-guardian-upgrade/README.md]: Fixed in cbecc39

File: tasks/eth/010-1-guardian-upgrade/NestedSignFromJson.s.sol (hunk 4 above)
Comment (mds1) [reply tasks/eth/010-1-guardian-upgrade/NestedSignFromJson.s.sol]: Ah you're right, done in 1bf8a4d

File: tasks/eth/010-1-guardian-upgrade/VALIDATION.md [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+
+Links:
+- [Etherscan](https://etherscan.io/address/0x95703e0982140D16f8ebA6d158FccEde42f04a4C)
+
+State Changes:
+
+- **Key:** 0xd30e835d3f35624761057ff5b27d558f97bd5be034621e62240e5c0b784abe68<br/>
+  **Before:** 0x0000000000000000000000009ba6e03d8b90de867373db8cf1a58d2f7f006b3a<br/>
+  **After:** 0x00000000000000000000000009f7150d8c019bef34450d6920f6b3608cefdaf2<br/>
+  **Meaning:** The Guardian address has been updated from `0x9BA6e03D8B90dE867373Db8cF1A58d2F7F006b3A` (Foundation Upgrades Safe) to `0x09f7150D8c019BeF34450d6920f6B3608ceFdAf2` (1/1 Safe owned by the Security Council).
```
Comment (mds1) [inline tasks/eth/010-1-guardian-upgrade/VALIDATION.md:133]:
```suggestion
  **Meaning:** The Guardian address has been updated from `0x9BA6e03D8B90dE867373Db8cF1A58d2F7F006b3A` (Foundation Operations Safe) to `0x09f7150D8c019BeF34450d6920f6B3608ceFdAf2` (1/1 Safe owned by the Security Council).
```

File: tasks/eth/010-2-sc-changes/SignFromJson.s.sol [hunk 7]
```diff
@@ -0,0 +1,330 @@
...
+
+    // Chains for this task.
+    string constant l1ChainName = "mainnet";
+    string constant l2ChainName = "op";
+
+    // Properties we want to verify.
+    uint256 constant expectedGuardLivenessInterval = 14 weeks;
+    uint256 constant expectedGuardMinOwners = 8;
+    uint256 constant expectedGuardThresholdPercentage = 75;
+    uint256 constant expectedThreshold = 10;
```
Comment (maurelian) [inline tasks/eth/010-2-sc-changes/SignFromJson.s.sol:65]: It would be good to add an `expectedOwnersCount` and assert that the owners are unchanged.

**Review by maurelian — COMMENTED** [review]
One request, then LGTM

File: tasks/eth/010-2-sc-changes/SignFromJson.s.sol (hunk 7 above)
Comment (mds1) [reply tasks/eth/010-2-sc-changes/SignFromJson.s.sol:65]: Done in 74ed190

**Review by maurelian — APPROVED** [review]

**Pull request merged**